src/app/view/components/dot-navigation/dot-navigation.component.ts

```diff
@@ -12,22 +12,37 @@ import { DotNavigationService } from './dot-navigation.service';
 })
 export class DotNavigationComponent implements OnInit {
     menu: Observable<DotMenu[]>;
+    @Output() onMenuLinkClicked = new EventEmitter<Event>();
 
-    constructor(private dotNavigationService: DotNavigationService) {}
+    constructor(private dotNavigationService: DotNavigationService, private sideNavEl: ElementRef) {}
 
     ngOnInit() {
         this.menu = this.dotNavigationService.items$;
     }
 
+    /**
+     * Respond to document events and collapse the sidenav if is clicked outside
+     * @param {*} event
+     * @memberof DotNavigationComponent
+     */
+    @HostListener('document:click', ['$event'])
```

Comment:
We should be listening to ```document:click``` in the ```main-legacy.component.ts``` because that's the one who can show and hide the sidenav, and there when a click happen in document we check the target... if the target is **not** the item that expand/collapse the accordion menus we hide the sidebar because it means the happen in something outside the sidenav OR in a sidenav link to navigate to another portlet.